### PR TITLE
create demo page showing how to use push in order to update a compone…

### PR DESCRIPTION
**Pull request by reiern70**
create demo page showing how to use push in order to update a component via web-sockets.

### CODE REVIEW

File: wicket-examples/src/main/java/org/apache/wicket/examples/websocket/progress/ProgressBarTogglePanel.java [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+
+        add(new AjaxLink<Void>("hideShowProgress")
+        {
+            @Override
+            public void onClick(AjaxRequestTarget target)
+            {
+                showProgress = !showProgress;
+                target.add(ProgressBarTogglePanel.this);
+            }
+        }.setBody(new IModel<String>()
```
Comment (martin-g) [inline wicket-examples/src/main/java/org/apache/wicket/examples/websocket/progress/ProgressBarTogglePanel.java:61]: You could use lambda here to make it less verbose.

File: wicket-examples/src/main/java/org/apache/wicket/examples/websocket/progress/ProgressUpdater.java [hunk 2]
```diff
@@ -0,0 +1,155 @@
...
+/**
+ * A helper class that uses the web connection to push components updates to the client.
+ */
+public class ProgressUpdater
+{
+	public static ProgressUpdateTask start(ConnectedMessage message, ScheduledExecutorService scheduledExecutorService)
+	{
+		// create an asynchronous task that will write the data to the client
+		ProgressUpdateTask progressUpdateTask = new ProgressUpdateTask(message.getApplication(), message.getSessionId(), message.getKey());
+        scheduledExecutorService.schedule(progressUpdateTask, 1, TimeUnit.SECONDS);
```
Comment (martin-g) [inline wicket-examples/src/main/java/org/apache/wicket/examples/websocket/progress/ProgressUpdater.java:41]: indentation

File: wicket-examples/src/main/java/org/apache/wicket/examples/websocket/progress/ProgressUpdater.java [hunk 3]
```diff
@@ -0,0 +1,155 @@
...
+					if (connection == null || !connection.isOpen())
+					{
+						running = false;
+						// stop if the web socket connection is closed
+						return;
+					}
+
+					if (canceled)
+					{
+						canceled = false;
```
Comment (martin-g) [inline wicket-examples/src/main/java/org/apache/wicket/examples/websocket/progress/ProgressUpdater.java:115]: why ?

File: wicket-examples/src/main/java/org/apache/wicket/examples/websocket/progress/ProgressUpdater.java [hunk 4]
```diff
@@ -0,0 +1,155 @@
...
+					progress++;
+				}
+				catch (InterruptedException x)
+				{
+					Thread.currentThread().interrupt();
+					break;
+				}
+				catch (Exception e)
+				{
+					e.printStackTrace();
```
Comment (martin-g) [inline wicket-examples/src/main/java/org/apache/wicket/examples/websocket/progress/ProgressUpdater.java:135]: Use SLF4J

File: wicket-examples/src/main/java/org/apache/wicket/examples/websocket/JSR356Application.java [hunk 5]
```diff
@@ -50,6 +50,7 @@ public void init()
 		setRootRequestMapper(new HttpsMapper(getRootRequestMapper(), new HttpsConfig(8080, 8443)));
 
 		mountPage("/behavior", WebSocketBehaviorDemoPage.class);
+		mountPage("/progress", WebSocketPushUpdateProgressDemoPage.class);
```
[the remark this answers is not in the document]
Comment (reiern70) [reply wicket-examples/src/main/java/org/apache/wicket/examples/websocket/JSR356Application.java:53]: This is a page where server pushes changes to update a progress "bar".  How to call it?

File: wicket-examples/src/main/java/org/apache/wicket/examples/websocket/WebSocketPushUpdateProgressDemoPage.java [hunk 6]
```diff
@@ -0,0 +1,30 @@
...
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.wicket.examples.websocket;
+
+import org.apache.wicket.examples.WicketExamplePage;
+import org.apache.wicket.examples.websocket.progress.ProgressBarTogglePanel;
+import org.apache.wicket.protocol.https.RequireHttps;
+
+@RequireHttps
```
[the remark this answers is not in the document]
Comment (reiern70) [reply wicket-examples/src/main/java/org/apache/wicket/examples/websocket/WebSocketPushUpdateProgressDemoPage.java:23]: The same annotation is on other classes on package. I even didn't think about the reason

File: wicket-examples/src/main/java/org/apache/wicket/examples/websocket/progress/ProgressUpdater.java (hunk 3 above)
Comment (reiern70) [reply wicket-examples/src/main/java/org/apache/wicket/examples/websocket/progress/ProgressUpdater.java:115]: Task can be restarted... I want canceled back to false. This is just to have a way for sending a command to task (in next iteration just finish)

File: wicket-examples/src/main/java/org/apache/wicket/examples/websocket/progress/ProgressUpdater.java (hunk 4 above)
Comment (reiern70) [reply wicket-examples/src/main/java/org/apache/wicket/examples/websocket/progress/ProgressUpdater.java:135]: Copy and paste from ChartUdater. I will fix.

File: wicket-examples/src/main/java/org/apache/wicket/examples/websocket/JSR356Application.java (hunk 5 above)
[the remark this answers is not in the document]
Comment (martin-g) [reply wicket-examples/src/main/java/org/apache/wicket/examples/websocket/JSR356Application.java:53]: I think `/push` would be a better path.
We, Wicket devs, know this functionality as `WebSocket Push`.

File: wicket-examples/src/main/java/org/apache/wicket/examples/websocket/WebSocketPushUpdateProgressDemoPage.java (hunk 6 above)
[the remark this answers is not in the document]
Comment (martin-g) [reply wicket-examples/src/main/java/org/apache/wicket/examples/websocket/WebSocketPushUpdateProgressDemoPage.java:23]: Hm. OK then :-)

File: wicket-examples/src/main/java/org/apache/wicket/examples/websocket/HomePage.html [hunk 7]
```diff
@@ -29,6 +29,7 @@
 					<li><a href="WebSocketBehaviorDemoPage.html">demo with WebSocketBehavior</a></li>
 					<li><a href="WebSocketResourceDemoPage.html">demo with WebSocketResource</a></li>
 					<li><a href="WebSocketMultiTabResourceDemoPage.html">demo with WebSocketResource and multiple tabs</a></li>
+					<li><a href="WebSocketPushUpdateProgressDemoPage.html">Update a component via push notifications</a></li>
```
[the remark this answers is not in the document]
Comment (reiern70) [reply wicket-examples/src/main/java/org/apache/wicket/examples/websocket/HomePage.html:32]: done

File: wicket-examples/src/main/java/org/apache/wicket/examples/websocket/JSR356Application.java (hunk 5 above)
[the remark this answers is not in the document]
Comment (reiern70) [reply wicket-examples/src/main/java/org/apache/wicket/examples/websocket/JSR356Application.java:53]: Done

File: wicket-examples/src/main/java/org/apache/wicket/examples/websocket/progress/ProgressBarTogglePanel.java (hunk 1 above)
Comment (reiern70) [reply wicket-examples/src/main/java/org/apache/wicket/examples/websocket/progress/ProgressBarTogglePanel.java:61]: done.

File: wicket-examples/src/main/java/org/apache/wicket/examples/websocket/progress/ProgressUpdater.java (hunk 2 above)
Comment (reiern70) [reply wicket-examples/src/main/java/org/apache/wicket/examples/websocket/progress/ProgressUpdater.java:41]: done

File: wicket-examples/src/main/java/org/apache/wicket/examples/websocket/progress/ProgressUpdater.java (hunk 4 above)
Comment (reiern70) [reply wicket-examples/src/main/java/org/apache/wicket/examples/websocket/progress/ProgressUpdater.java:135]: done.

**Comment by reiern70** [comment]
@martin-g thanks for review. Implemented suggested changes

**Comment by reiern70** [comment]
Superseded by

https://github.com/apache/wicket/pull/479

**Pull request closed without merging**